my-app/src/App.js

```diff
@@ -1,14 +1,54 @@
 import React, { Component } from 'react';
-import logo from './logo.svg';
 import './App.css';
-import FirstComponent from './FirstComponent'
 import './assets/styles/layout.css';
+import TodoList from './TodoList';
 
 class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todoArray: []
+    };
+  this.inputData = React.createRef();
+  this.addItem = this.addItem.bind(this);
+  this.deleteItem = this.deleteItem.bind(this);
+  }
+
+  addItem(event) {
+    event.preventDefault();
+    let insertElement = this.inputData.current;
+    let insertValue = insertElement.value;
+    let tempArray = this.state.todoArray;
```

Comment:
this will create a reference to this.state.todoArray. so create a copy of it like 
`let tempArray = [...this.state.todoArray];`